tests/load/pipeline/test_filesystem_pipeline.py

```diff
@@ -1254,6 +1255,64 @@ def my_resource_inc(prim_key=dlt.sources.incremental("id")):
     assert load_table_counts(p, "items") == {"items": 4 if restore else 6}
 
 
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(all_buckets_filesystem_configs=True),
+    ids=lambda x: x.name,
+)
+def test_max_state_files_exception(destination_config: DestinationTestConfiguration) -> None:
+    os.environ["DESTINATION__FILESYSTEM__MAX_STATE_FILES"] = "1"
+
+    with pytest.raises(Exception, match="The max_state_files must be grater than 1."):
+        p = destination_config.setup_pipeline(
+            "p1", dataset_name=destination_config.destination_name
+        )
+        p.run([1, 2, 3])
+
+
+@pytest.mark.parametrize(
```

Comment:
I would parametrize DESTINATION__FILESYSTEM__MAX_STATE_FILES to [None, 10, 30] so see that the settings works